Marlin/src/lcd/dwin/e3v2/dwin.cpp

```diff
@@ -8854,7 +8859,12 @@ duration_t estimate_remaining_time(const duration_t elapsed)
   // _speed = (_fileFraction * (float)card.getFileSize()) / _elapsedTime;
   // _remainSize = (float)card.getFileSize() * (1 - _fileFraction);
   // _remain_time = _speed * _remainSize;
-  return duration_t((((_card_percent * 0.01f) * (float)card.getFileSize()) / ((elapsed.value + 1) - dwin_heat_time)) * ((float)card.getFileSize() * (100 - _card_percent)));
+  if (elapsed.value - dwin_heat_time < 60 && _card_percent < 1) {
+    return duration_t(0);
+  }
+
+  // remaining time is (elapsed time / relative file position) - elapsed time
+  return duration_t((elapsed.value - dwin_heat_time) / (_card_percent * 0.01f + (_card_percent > 0 ? 0.001f : 0)) - (elapsed.value - dwin_heat_time));
```

Comment:
this probable is should utilize the `PROGRESS_SCALE` to get estimations sooner than once 1% is hit. The `_card_percent > 0 ? 0.001f : 0` bit is to avoid division by 0.